rspace/src/main/scala/coop/rchain/rspace/HotStore.scala

```diff
@@ -142,6 +144,12 @@ private class InMemHotStore[F[_]: Sync, C, P, A, K](
     } yield ()
 
   def getJoins(channel: C): F[Seq[Seq[C]]] =
+    for {
+      cached <- getCachedJoins(channel)
+      state  <- S.read
+    } yield (state.installedJoins.get(channel).getOrElse(Seq.empty) ++: cached)
+
+  private[this] def getCachedJoins(channel: C): F[Seq[Seq[C]]] =
```

Comment:
nit: not convinced this is the best name

Reply:
As agreed, renamed to `internalGet...`